bigtable: TestIntegration_ExportBuiltInMetrics failed

Note: #13732 was also for this test, but it was closed more than 10 days ago. So, I didn't mark it flaky.

----

commit: c7607be52757b803df345670b5d0621c2bb9ba30
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/fb3602a5-5663-4443-8b76-57d66d01017e), [Sponge](http://sponge2/fb3602a5-5663-4443-8b76-57d66d01017e)
status: failed

Error from logs:
```go
=== RUN   TestIntegration_ExportBuiltInMetrics
    integration_test.go:542: bigtable_connpool:  connection factory prime failed: factory prime failed after 3 attempts: rpc error: code = FailedPrecondition desc = Table currently being created: projects/dulcet-port-762/instances/gc-bt-it-instance/tables/mytable-1781739762957794772-08
--- FAIL: TestIntegration_ExportBuiltInMetrics (241.85s)
```

--- a/bigtable/admin/apiv2/table_admin_integration_test.go
+++ b/bigtable/admin/apiv2/table_admin_integration_test.go
@@ -22,6 +22,7 @@ import (
 	"time"
 
 	adminpb "cloud.google.com/go/bigtable/admin/apiv2/adminpb"
+	"cloud.google.com/go/internal/uid"
 	"google.golang.org/api/option"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/codes"
@@ -36,6 +37,13 @@ type testEnv struct {
 	instance string
 	cluster  string
 }
+
+var (
+	sourceTableSpace   = uid.NewSpace("it-src-table", &uid.Options{Short: true})
+	backupSpace        = uid.NewSpace("it-backup", &uid.Options{Short: true})
+	restoredTableSpace = uid.NewSpace("it-restored-table", &uid.Options{Short: true})
+	replTestTableSpace = uid.NewSpace("it-repl-table", &uid.Options{Short: true})
+)
 
 func setupIntegration(t *testing.T) *testEnv {
 	t.Helper()
@@ -94,10 +102,9 @@ func TestIntegration_RestoreTable(t *testing.T) {
 	ctx := context.Background()
 	client := env.client
 
-	suffix := time.Now().Format("20060102-150405")
-	sourceTableID := fmt.Sprintf("src-table-%s", suffix)
-	backupID := fmt.Sprintf("backup-%s", suffix)
-	restoredTableID := fmt.Sprintf("restored-table-%s", suffix)
+	sourceTableID := sourceTableSpace.New()
+	backupID := backupSpace.New()
+	restoredTableID := restoredTableSpace.New()
 
 	instancePath := fmt.Sprintf("projects/%s/instances/%s", env.project, env.instance)
 	sourceTablePath := fmt.Sprintf("%s/tables/%s", instancePath, sourceTableID)
@@ -178,8 +185,7 @@ func TestIntegration_WaitForConsistency(t *testing.T) {
 	ctx := context.Background()
 	client := env.client
 
-	suffix := time.Now().Format("20060102-150405")
-	tableID := fmt.Sprintf("repl-test-table-%s", suffix)
+	tableID := replTestTableSpace.New()
 	instancePath := fmt.Sprintf("projects/%s/instances/%s", env.project, env.instance)
 	tablePath := fmt.Sprintf("%s/tables/%s", instancePath, tableID)
 

--- a/bigtable/admin/apiv2/table_admin_integration_test.go
+++ b/bigtable/admin/apiv2/table_admin_integration_test.go
@@ -94,12 +94,14 @@ func setupIntegration(t *testing.T) *testEnv {
 }
 
 func TestIntegration_RestoreTable(t *testing.T) {
+	t.Parallel()
 	env := setupIntegration(t)
 	if os.Getenv("BIGTABLE_EMULATOR_HOST") == "" && env.cluster == "" {
 		t.Skip("Missing GCLOUD_TESTS_BIGTABLE_CLUSTER for non-emulator run")
 	}
 
-	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	t.Cleanup(cancel)
 	client := env.client
 
 	sourceTableID := sourceTableSpace.New()
@@ -180,9 +182,11 @@ func TestIntegration_RestoreTable(t *testing.T) {
 }
 
 func TestIntegration_WaitForConsistency(t *testing.T) {
+	t.Parallel()
 	env := setupIntegration(t)
 
-	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	t.Cleanup(cancel)
 	client := env.client
 
 	tableID := replTestTableSpace.New()

--- a/bigtable/admin/apiv2/table_admin_integration_test.go
+++ b/bigtable/admin/apiv2/table_admin_integration_test.go
@@ -102,6 +102,7 @@ func TestIntegration_RestoreTable(t *testing.T) {
 
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
 	t.Cleanup(cancel)
+	cleanupCtx := context.WithoutCancel(ctx)
 	client := env.client
 
 	sourceTableID := sourceTableSpace.New()
@@ -124,7 +125,7 @@ func TestIntegration_RestoreTable(t *testing.T) {
 		t.Fatalf("Failed to create source table: %v", err)
 	}
 	t.Cleanup(func() {
-		client.DeleteTable(ctx, &adminpb.DeleteTableRequest{Name: sourceTablePath})
+		client.DeleteTable(cleanupCtx, &adminpb.DeleteTableRequest{Name: sourceTablePath})
 	})
 
 	// 2. Create backup
@@ -145,7 +146,7 @@ func TestIntegration_RestoreTable(t *testing.T) {
 		t.Fatalf("Failed to initiate backup: %v", err)
 	}
 	t.Cleanup(func() {
-		client.DeleteBackup(ctx, &adminpb.DeleteBackupRequest{Name: backupPath})
+		client.DeleteBackup(cleanupCtx, &adminpb.DeleteBackupRequest{Name: backupPath})
 	})
 
 	_, err = opCreateBackup.Wait(ctx)
@@ -165,7 +166,7 @@ func TestIntegration_RestoreTable(t *testing.T) {
 		t.Fatalf("RestoreTable failed: %v", err)
 	}
 	t.Cleanup(func() {
-		client.DeleteTable(ctx, &adminpb.DeleteTableRequest{Name: restoredTablePath})
+		client.DeleteTable(cleanupCtx, &adminpb.DeleteTableRequest{Name: restoredTablePath})
 	})
 
 	// 4. Verify restored table exists
@@ -187,6 +188,7 @@ func TestIntegration_WaitForConsistency(t *testing.T) {
 
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
 	t.Cleanup(cancel)
+	cleanupCtx := context.WithoutCancel(ctx)
 	client := env.client
 
 	tableID := replTestTableSpace.New()
@@ -203,7 +205,7 @@ func TestIntegration_WaitForConsistency(t *testing.T) {
 		t.Fatalf("Failed to create table: %v", err)
 	}
 	t.Cleanup(func() {
-		client.DeleteTable(ctx, &adminpb.DeleteTableRequest{Name: tablePath})
+		client.DeleteTable(cleanupCtx, &adminpb.DeleteTableRequest{Name: tablePath})
 	})
 
 	// Wait for replication

--- a/bigtable/integration_test.go
+++ b/bigtable/integration_test.go
@@ -537,7 +537,7 @@ func TestIntegration_ExportBuiltInMetrics(t *testing.T) {
 		t.Skip("Skip long running tests in short mode or non-prod environments")
 	}
 
-	client, err := testEnv.NewClient()
+	client, err := newClientWithRetry(ctx, testEnv)
 	if err != nil {
 		t.Fatal(err)
 	}

